gdx/src/com/badlogic/gdx/utils/Pools.java

```diff
@@ -16,104 +16,126 @@
-	static private final ObjectMap<Class<?>, Pool<?>> supplierPoolsCache = new ObjectMap<>();
+	static public boolean WARN_ON_REFLECTION_POOL_CREATION = true;
+	static public boolean THROW_ON_REFLECTION_POOL_CREATION = false;
+
+	static {
+		set(Array::new);
+		set(ChangeEvent::new);
+		set(DebugRect::new);
+		set(FocusEvent::new);
+		set(GlyphLayout::new);
+		set(GlyphRun::new);
+		set(HttpRequest::new);
+		set(InputEvent::new);
+		set(Rectangle::new);
+		set(TouchFocus::new);
+
+		// Actions
+		set(AddAction::new);
+		set(AddListenerAction::new);
+		set(AfterAction::new);
+		set(AlphaAction::new);
+		set(ColorAction::new);
+		set(DelayAction::new);
+		set(FloatAction::new);
+		set(IntAction::new);
+		set(LayoutAction::new);
+		set(MoveByAction::new);
+		set(MoveToAction::new);
+		set(ParallelAction::new);
+		set(RemoveAction::new);
+		set(RemoveActorAction::new);
+		set(RemoveListenerAction::new);
+		set(RepeatAction::new);
+		set(RotateByAction::new);
+		set(RotateToAction::new);
+		set(RunnableAction::new);
+		set(ScaleByAction::new);
+		set(ScaleToAction::new);
+		set(SequenceAction::new);
+		set(SizeByAction::new);
+		set(SizeToAction::new);
+		set(TimeScaleAction::new);
+		set(TouchableAction::new);
+		set(VisibleAction::new);
+	}
 
 	/** Returns a new or existing pool for the specified type, stored in a Class to {@link Pool} map. Note the max size is ignored
-	 * if this is not the first time this pool has been requested.
-	 *
-	 * @deprecated Use {@link Pools#get(PoolSupplier, int)} instead */
-	@Deprecated
+	 * if this is not the first time this pool has been requested. */
 	static public <T> Pool<T> get (Class<T> type, int max) {
 		Pool pool = typePools.get(type);
 		if (pool == null) {
+			if (THROW_ON_REFLECTION_POOL_CREATION) throw new RuntimeException(
+				"Please manually define a Pool for " + type + " by calling Pools#set before calling Pools#get");
+			if (WARN_ON_REFLECTION_POOL_CREATION && Gdx.app != null) Gdx.app.error("Pools",
+				"Please manually define a Pool for " + type + " by calling Pools#set before calling Pools#get");
 			pool = new ReflectionPool(type, 4, max);
 			typePools.put(type, pool);
 		}
 		return pool;
 	}
 
 	/** Returns a new or existing pool for the specified type, stored in a Class to {@link Pool} map. The max size of the pool used
-	 * is 100.
-	 *
-	 * @deprecated Use {@link Pools#get(PoolSupplier)} instead */
-	@Deprecated
+	 * is 100. */
 	static public <T> Pool<T> get (Class<T> type) {
 		return get(type, 100);
 	}
 
-	/** Returns a new or existing pool for the specified type, stored in a Class to {@link Pool} map. Note the max size is ignored
-	 * if this is not the first time this pool has been requested. This method may create a temporary object once, but the
-	 * allocation will probably be eliminated on most platforms. <br>
-	 * Usage can use java 8 method references: {@code Pools.get(MyClass::new, max)} */
-	static public <T> Pool<T> get (PoolSupplier<T> poolTypeSupplier, int max) {
```

Comment:
It may have only been in for one version, but this was still the only non-deprecated API to use Pools in 1.13.5. I think it's a bad idea to remove it wholesale, without warning.

Reply:
Sure, I can readd it with a Deprecation warning, I just figure it would be simple to migrate back for the few who already have migrated theirs. Since they already did it once I mean.

Reply:
I wouldn't readd with deprecation tbh (as long as we release in a short period of time of course), for the few users who may have made the change it should be simple to migrate back.